### PR TITLE
Add SeatFilter API

### PR DESCRIPTION
**Pull request by vonox7**

### CODE REVIEW

**Review by functionaldude — COMMENTED** [review]
We should deprecate `receiveJsonMap()` and `receiveJsonStringMap()` and only use the ClientPayloads

**Review by functionaldude — CHANGES_REQUESTED** [review]
We should not introduce `receiveJsonStringMap()` if we are going to deprecate it.

**Comment by vonox7** [comment]
Yes. But lets make an own PR out of that, refactoring everything is out of scope of this PR

**Review by functionaldude — APPROVED** [review]

File: src/serverMain/kotlin/com/studo/campusqr/database/Payloads.kt [hunk 1]
```diff
@@ -76,6 +76,21 @@ class BackendAccess : MongoMainEntry(), ClientPayloadable<ClientAccessManagement
   override fun toClientClass(language: String) = throw NotImplementedError()
 }
 
+class BackendSeatFilter : MongoMainEntry(), ClientPayloadable<ClientSeatFilter> {
+  lateinit var locationId: String
+  lateinit var editedBy: String // userId
+  lateinit var lastEditDate: Date
+  var seat: Int = 0
```
Comment (ffactory-ofcl) [inline src/serverMain/kotlin/com/studo/campusqr/database/Payloads.kt:83]: Why give it a default?

File: src/serverMain/kotlin/com/studo/campusqr/endpoints/LocationManagement.kt [hunk 2]
```diff
@@ -211,6 +218,7 @@ suspend fun AuthenticatedApplicationCall.deleteLocation() {
     getCollection<BackendLocation>().deleteOne(BackendLocation::_id equal locationId)
     getCollection<CheckIn>().deleteMany(CheckIn::locationId equal locationId)
     getCollection<BackendAccess>().deleteMany(BackendAccess::locationId equal locationId)
+    getCollection<BackendSeatFilter>().deleteMany(BackendAccess::locationId equal locationId)
```
Comment (ffactory-ofcl) [inline src/serverMain/kotlin/com/studo/campusqr/endpoints/LocationManagement.kt:221]:
```suggestion
    getCollection<BackendSeatFilter>().deleteMany(BackendSeatFilter::locationId equal locationId)
```

**Review by zigavehovec — APPROVED** [review]

**Review by ffactory-ofcl — CHANGES_REQUESTED** [review]

File: src/serverMain/kotlin/com/studo/campusqr/database/Payloads.kt (hunk 1 above)
Comment (vonox7) [reply src/serverMain/kotlin/com/studo/campusqr/database/Payloads.kt:83]: kotlin limitation: primitives can't be lateinit

**Review by ffactory-ofcl — APPROVED** [review]

**Pull request merged**